src/Generators/TestsGenerator.php

```diff
@@ -145,30 +146,32 @@ protected function buildEntityObject($model)
         return $result;
     }
 
-    protected function getModelClass($model)
+    protected function getModelClass($model): string
     {
-        return "App\\Models\\{$model}";
+        $modelNamespace = $this->getNamespace('models');
+        return "{$modelNamespace}\\{$model}";
     }
 
-    protected function getModelFields($model)
+    protected function getModelFields($model): array
     {
         $modelClass = $this->getModelClass($model);
 
         return $this->filterBadModelField($modelClass::getFields());
     }
 
-    protected function getMockModel($model)
+    protected function getMockModel($model): array
     {
         $modelClass = $this->getModelClass($model);
-
-        $factory = (method_exists($modelClass, 'factory')) ? $modelClass::factory() : factory($modelClass);
+        $useModelClassFactory = method_exists($modelClass, 'factory')
+            && class_exists(Factory::resolveFactoryName($modelClass));
+        $factory = ($useModelClassFactory) ? $modelClass::factory() : factory($modelClass);
```

Comment:
it seems we need to use the same logic as in `isFactoryExists` method, could we implement one method which will contains one logic and use it here